microcloud/cmd/microcloud/main_init.go

```diff
@@ -179,14 +179,12 @@ func lookupPeers(s *service.Handler, autoSetup bool, iface *net.Interface, subne
 		}()
 	}
 
-	var timeAfter <-chan time.Time
+	timeoutDuration := time.Minute
```

Comment:
Why dont you use a context.WithTimeout() here then you can pass it to  mdns.LookupPeers() too and potentially just wait for that to return.

Reply:
fixed